packages/swapper/src/swappers/thorchain/utils/getInboundAddressDataForChain.ts

```diff
@@ -0,0 +1,17 @@
+import { adapters, AssetId } from '@shapeshiftoss/caip'
+
+import type { InboundAddressResponse } from '../types'
+import { thorService } from './thorService'
+
+export const getInboundAddressDataForChain = async (
+  daemonUrl: string,
+  assetId: AssetId | undefined,
+): Promise<InboundAddressResponse | undefined> => {
+  if (!assetId) return undefined
+  const assetPoolId = adapters.assetIdToPoolAssetId({ assetId })
+  const assetChainSymbol = assetPoolId?.slice(0, assetPoolId.indexOf('.'))
+  const { data: inboundAddresses } = await thorService.get<InboundAddressResponse[]>(
+    `${daemonUrl}/lcd/thorchain/inbound_addresses`,
+  )
+  return inboundAddresses.find((inbound) => inbound.chain === assetChainSymbol)
```

Comment:
we should filter/exclude any address that has `"halted": true` property

Reply:
For sure. Actioned in https://github.com/shapeshift/lib/pull/1080/commits/199015a45ca18acc92337e5b305d8712f264203a.